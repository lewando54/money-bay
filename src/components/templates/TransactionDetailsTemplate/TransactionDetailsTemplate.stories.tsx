/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { Text,View } from 'react-native'
import { Meta,StoryObj } from '@storybook/react-native'
import { TCurrency } from 'src/utils/currency'

import Images from '@assets/images'

import TransactionDetailsTemplate from './TransactionDetailsTemplate'

import globalThemeStyle, { globalColors } from '@styling/GlobalTheme.style'

interface IVisibleMockupProps {
    wholeAmount?: string
    decimalAmount?: string
    name?: string
    date?: Date
    recipientName?: string
    statusElement?: React.ReactNode
    cardNumber?: string
    amount?: number
    fee?: number
    residualBalance?: number
    currency?: TCurrency
    onDownloadPDFPress?: () => void
}

function Visible({
    wholeAmount,
    decimalAmount,
    name,
    date,
    recipientName,
    statusElement,
    cardNumber,
    amount,
    fee,
    residualBalance,
    currency,
    onDownloadPDFPress
}: IVisibleMockupProps){
    return (
        <View style={{flex: 1, backgroundColor: 'white'}}>
            <TransactionDetailsTemplate
                wholeAmount={wholeAmount}
                decimalAmount={decimalAmount}
                name={name}
                date={date}
                recipientName={recipientName}
                statusElement={statusElement}
                cardNumber={cardNumber}
                amount={amount}
                fee={fee}
                residualBalance={residualBalance}
                currency={currency}
                onDownloadPDFPress={onDownloadPDFPress}
            />
        </View>
    )
}

const meta = {
    component: Visible,
    argTypes: {
        wholeAmount: {control: 'text'},
        decimalAmount: {control: 'text'},
        name: {control: 'text'},
        date: {control: 'date'},
        recipientName: {control: 'text'},
        statusElement: {control: 'object'},
        cardNumber: {control: 'text'},
        amount: {control: 'number'},
        fee: {control: 'number'},
        residualBalance: {control: 'number'},
        currency: {control: 'object'},
        onDownloadPDFPress: {action: 'Download PDF Pressed'},
    }
} satisfies Meta<typeof Visible>

export default meta
type Story = StoryObj<typeof Visible>

export const Default: Story = {
    args: {
        wholeAmount: '10',
        decimalAmount: '0',
        name: 'Mobile Payment',
        date: new Date('2022-09-10T11:34:13'),
        recipientName: 'Hillary Holmes',
        statusElement: <View style={{flexDirection: 'row', gap: 5, alignItems: 'center'}}><Images.successDetailsSVG/><Text style={{...globalThemeStyle.text_SemiBold, color: globalColors.GOOD_GREEN, fontSize: 14}}>Success</Text></View>,
        cardNumber: '**** 4253',
        fee: 0,
        residualBalance: 4853.27,
        currency: {iso: 'usd', symbol: '$', left: true}
    }
}
