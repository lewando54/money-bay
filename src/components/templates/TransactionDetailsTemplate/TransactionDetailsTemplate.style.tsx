import { Dimensions, StyleSheet } from 'react-native'

import globalThemeStyle, { BODY_TEXT_COLOR, MAIN_DARK }  from '../../../styling/GlobalTheme.style'

const {width, height} = Dimensions.get('screen')

export default StyleSheet.create({
    background: {
        width: width,
        resizeMode: 'contain',
        maxHeight: width - 27,
        position: 'absolute'
    },
    mainContentContainer: {
        flex: 1,
        alignItems: 'center',
        marginTop: height * 0.1034,
        gap: height * 0.0123
    },
    dateText: {
        ...globalThemeStyle.text_Regular,
        fontSize: 12,
        color: BODY_TEXT_COLOR
    },
    moneyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end'
    },
    bigMoneyText: {
        ...globalThemeStyle.text_Bold,
        color: MAIN_DARK,
        fontSize: 28
    },
    smallMoneyText: {
        ...globalThemeStyle.text_Bold,
        color: MAIN_DARK,
        fontSize: 16,
        marginBottom: height * 0.004
    },
    tableContainer: {
        paddingHorizontal: width * 0.0533,
        width: '100%',
        flex: 1,
        marginBottom: height * 0.0246
    },
    cellContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        width: '100%',
        paddingVertical: height * 0.0246,
        borderBottomColor: '#CED6E1',
        borderBottomWidth: 1
    },
    cellHeader: {
        ...globalThemeStyle.text_Regular,
        fontSize: 16,
        color: BODY_TEXT_COLOR
    },
    cellData: {
        ...globalThemeStyle.text_SemiBold,
        fontSize: 16,
        color: MAIN_DARK
    },
    downloadButton: {
        marginTop: 'auto',
    }
})
