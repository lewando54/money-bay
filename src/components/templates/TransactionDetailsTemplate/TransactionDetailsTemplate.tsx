import React from 'react'
import { Image, Text,View } from 'react-native'
import Images from '@assets/images'

import { TCurrency } from '../../../utils/currency'
import Button from '../../atoms/Button/Button'
import Separator from '../../atoms/Separator/Separator'
import ButtonCircleWithText from '../../molecules/ButtonCircleWithText/ButtonCircleWithText'

import globalThemeStyle, { GOOD_GREEN } from '../../../styling/GlobalTheme.style'
import { ORANGE } from '../../../styling/GlobalTheme.style'
import transactionDetailsTemplateStyle from './TransactionDetailsTemplate.style'

interface ITransactionDetailsTemplateProps {
    wholeAmount?: string
    decimalAmount?: string
    name?: string
    date?: Date
    recipientName?: string
    statusElement?: React.ReactNode
    cardNumber?: string
    amount?: number
    fee?: number
    residualBalance?: number
    currency?: TCurrency
    onDownloadPDFPress?: () => void
    testID?: string
}

export default function TransactionDetailsTemplate({
    wholeAmount='10',
    decimalAmount='0',
    name='Mobile Payment',
    date=new Date('2022-09-10T11:34:13'),
    recipientName='Hillary Holmes',
    statusElement=<View style={{flexDirection: 'row', gap: 5, alignItems: 'center'}}><Images.successDetailsSVG/><Text style={{...globalThemeStyle.text_SemiBold, color: GOOD_GREEN, fontSize: 14}}>Success</Text></View>,
    cardNumber='**** 4253',
    fee=0,
    residualBalance=4853.27,
    currency={iso: 'usd', symbol: '$', left: true},
    onDownloadPDFPress,
    testID
}: ITransactionDetailsTemplateProps){
    return (
        <View testID={testID} style={{flex: 1}}>
            <Image source={Images.bgSignIn} style={transactionDetailsTemplateStyle.background} />
            <View style={transactionDetailsTemplateStyle.mainContentContainer}>
                <ButtonCircleWithText circleBackgroundColor={ORANGE} text={name} icon={<Images.smartphoneSVG color={'white'}/>}/>
                <Text style={transactionDetailsTemplateStyle.dateText}>{date.toLocaleDateString('en-US', {year: 'numeric', month: 'short', day: 'numeric'})} at {date.toLocaleTimeString('en-US', {hourCycle: 'h12', hour: 'numeric', minute: 'numeric'})}</Text>
                <View style={transactionDetailsTemplateStyle.moneyContainer}>
                    <Text style={transactionDetailsTemplateStyle.bigMoneyText}>- {currency.symbol} {wholeAmount}.</Text>
                    <Text style={transactionDetailsTemplateStyle.smallMoneyText}>{parseInt(decimalAmount).toLocaleString('en-US', {minimumIntegerDigits: 2})}</Text>
                </View>
                <Text style={transactionDetailsTemplateStyle.cellHeader}>Sent to {recipientName}</Text>
                {statusElement}
                <Separator />
                <View style={transactionDetailsTemplateStyle.tableContainer}>
                    <View style={transactionDetailsTemplateStyle.cellContainer}>
                        <Text style={transactionDetailsTemplateStyle.cellHeader}>Sent to</Text>
                        <Text style={transactionDetailsTemplateStyle.cellData}>{recipientName}</Text>
                    </View>
                    <View style={transactionDetailsTemplateStyle.cellContainer}>
                        <Text style={transactionDetailsTemplateStyle.cellHeader}>Card</Text>
                        <Text style={transactionDetailsTemplateStyle.cellData}>{cardNumber}</Text>
                    </View>
                    <View style={transactionDetailsTemplateStyle.cellContainer}>
                        <Text style={transactionDetailsTemplateStyle.cellHeader}>Amount</Text>
                        <Text style={transactionDetailsTemplateStyle.cellData}>{wholeAmount+'.'+decimalAmount} {currency.iso.toLocaleUpperCase()}</Text>
                    </View>
                    <View style={transactionDetailsTemplateStyle.cellContainer}>
                        <Text style={transactionDetailsTemplateStyle.cellHeader}>Fee</Text>
                        <Text style={transactionDetailsTemplateStyle.cellData}>{fee} {currency.iso.toLocaleUpperCase()}</Text>
                    </View>
                    <View style={transactionDetailsTemplateStyle.cellContainer}>
                        <Text style={transactionDetailsTemplateStyle.cellHeader}>Residual balance</Text>
                        <Text style={transactionDetailsTemplateStyle.cellData}>{residualBalance} {currency.iso.toLocaleUpperCase()}</Text>
                    </View>
                    <Button containerStyle={transactionDetailsTemplateStyle.downloadButton} onClick={onDownloadPDFPress}>Download PDF</Button>
                </View>
            </View>
        </View>
    )
}
