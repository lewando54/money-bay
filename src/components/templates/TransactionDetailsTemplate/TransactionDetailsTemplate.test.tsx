import React from 'react'
import renderer from 'react-test-renderer'
import TransactionDetailsTemplate from './TransactionDetailsTemplate'
describe('TransactionDetailsTemplate', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<TransactionDetailsTemplate />)
        expect(tree).toMatchSnapshot()
    })
})
