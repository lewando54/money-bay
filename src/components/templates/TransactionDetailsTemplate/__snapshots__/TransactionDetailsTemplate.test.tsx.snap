// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetailsTemplate should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "maxHeight": 723,
        "position": "absolute",
        "resizeMode": "contain",
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 16.4082,
        "marginTop": 137.9356,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "elevation": 20,
          "gap": 10,
          "height": 95,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#1B1D4D",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
          "width": 120,
        }
      }
      testID="square-button-with-icon"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FF8A71",
              "borderRadius": 240,
              "height": 120,
              "justifyContent": "center",
              "width": 120,
            },
            {
              "elevation": 20,
              "shadowColor": "#16196670",
              "shadowOffset": {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
          ]
        }
      >
        <SvgMock
          color="white"
        />
      </View>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
              "textAlign": "center",
            },
          ]
        }
      >
        Mobile Payment
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      10 wrz 2022
       at 
      11:34 AM
    </Text>
    <View
      style={
        {
          "alignItems": "flex-end",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishBold",
            "fontSize": 28,
          }
        }
      >
        - 
        $
         
        10
        .
      </Text>
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishBold",
            "fontSize": 16,
            "marginBottom": 5.336,
          }
        }
      >
        00
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
        }
      }
    >
      Sent to 
      Hillary Holmes
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 5,
        }
      }
    >
      <SvgMock />
      <Text
        style={
          {
            "color": "#3EB290",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          }
        }
      >
        Success
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#CED6E1",
            "height": 1,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "flex": 1,
          "marginBottom": 32.8164,
          "paddingHorizontal": 39.975,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "borderBottomColor": "#CED6E1",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 32.8164,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            }
          }
        >
          Sent to
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 16,
            }
          }
        >
          Hillary Holmes
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#CED6E1",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 32.8164,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            }
          }
        >
          Card
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 16,
            }
          }
        >
          **** 4253
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#CED6E1",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 32.8164,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            }
          }
        >
          Amount
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 16,
            }
          }
        >
          10.0
           
          USD
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#CED6E1",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 32.8164,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            }
          }
        >
          Fee
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 16,
            }
          }
        >
          0
           
          USD
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#CED6E1",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 32.8164,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            }
          }
        >
          Residual balance
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 16,
            }
          }
        >
          4853.27
           
          USD
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginTop": "auto",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            {
              "colors": [
                4288142079,
                4279654301,
              ],
              "endPoint": [
                1,
                0.5,
              ],
              "locations": undefined,
              "startPoint": [
                0,
                0.5,
              ],
            }
          }
          style={
            [
              {
                "borderRadius": 10,
                "padding": 15,
              },
              false,
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 16,
                  "textAlign": "center",
                },
                {
                  "color": "white",
                },
                false,
                false,
              ]
            }
          >
            Download PDF
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;
